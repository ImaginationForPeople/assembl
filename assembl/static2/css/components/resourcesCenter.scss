.resources-center {

  .resource-block {
    margin: 100px 35px;

    .title-section {
      padding-bottom: 50px;
    }

    .resource-body {
      margin: 0 77px;

      img {
        max-height: 300px;
      }

      .margin-case-right {
        margin: 12px 50px 40px 0;
      }
      .margin-case-left {
        margin: 12px 0px 40px 50px;
      }

      .resource-text {
        width: 1200px;
        text-align: justify;
        font-size: $font-size-sm;
        line-height: 2;

        .resource-download-link {
          font-size: $font-size-m;
          margin-top: 20px;
        }
      }
    }
  }
}
