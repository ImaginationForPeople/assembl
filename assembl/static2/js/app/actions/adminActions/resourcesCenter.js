// @flow
import * as actionTypes from '../actionTypes';

export const createResource = (id: string, order: number): actionTypes.CreateResource => {
  return { id: id, order: order, type: actionTypes.CREATE_RESOURCE };
};

export const deleteResource = (id: string): actionTypes.DeleteResource => {
  return { id: id, type: actionTypes.DELETE_RESOURCE };
};

export const updateResourceDocument = (id: string, value: string): actionTypes.UpdateResourceDocument => {
  return {
    id: id,
    value: value,
    type: actionTypes.UPDATE_RESOURCE_DOCUMENT
  };
};

export const updateResourceEmbedCode = (id: string, value: string): actionTypes.UpdateResourceEmbedCode => {
  return {
    id: id,
    value: value,
    type: actionTypes.UPDATE_RESOURCE_EMBED_CODE
  };
};

export const updateResourceImage = (id: string, value: string): actionTypes.UpdateResourceImage => {
  return {
    id: id,
    value: value,
    type: actionTypes.UPDATE_RESOURCE_IMAGE
  };
};

export const updateResourceText = (id: string, locale: string, value: string): actionTypes.UpdateResourceText => {
  return {
    id: id,
    locale: locale,
    value: value,
    type: actionTypes.UPDATE_RESOURCE_TEXT
  };
};

export const updateResourceTitle = (id: string, locale: string, value: string): actionTypes.UpdateResourceTitle => {
  return {
    id: id,
    locale: locale,
    value: value,
    type: actionTypes.UPDATE_RESOURCE_TITLE
  };
};

export const updateResources = (resources: actionTypes.ResourcesArray): actionTypes.UpdateResources => {
  return {
    resources: resources,
    type: actionTypes.UPDATE_RESOURCES
  };
};

export const updateResourcesCenterPageTitle = (locale: string, value: string): actionTypes.UpdateRCPageTitle => {
  return {
    locale: locale,
    value: value,
    type: actionTypes.UPDATE_RC_PAGE_TITLE
  };
};

export const updateResourcesCenterHeaderImage = (value: File): actionTypes.UpdateResourcesCenterHeaderImage => {
  return {
    value: value,
    type: actionTypes.UPDATE_RC_HEADER_IMAGE
  };
};

export const updateResourcesCenterPage = (
  titleEntries: Array<any>,
  headerImage: File | null
): actionTypes.UpdateResourcesCenterPage => {
  return {
    headerImage: headerImage,
    titleEntries: titleEntries,
    type: actionTypes.UPDATE_RC_PAGE
  };
};