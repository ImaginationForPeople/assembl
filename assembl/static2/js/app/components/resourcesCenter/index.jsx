// @flow
import React from 'react';
import Header from '../common/header';
import ResourceBlock from './resourceBlock';

export type Media = {
  type: string,
  url: string
};

export type Resource = {
  title: string,
  text: string,
  image: Object,
  embedCode: string,
  doc: Object
};

type ResourcesCenterProps = {
  headerTitle: string,
  resources: Array<Resource>,
  headerBackgroundUrl: string
};

const ResourcesCenter = ({ resources, headerBackgroundUrl, headerTitle }: ResourcesCenterProps) => {
  return (
    <div className="resources-center">
      <Header title={headerTitle} imgUrl={headerBackgroundUrl} />
      <section>
        {resources.map((resource, index) => {
          const { title, text, embedCode, image, doc } = resource;
          return (
            <ResourceBlock title={title} text={text} embedCode={embedCode} image={image} doc={doc} index={index} key={index} />
          );
        })}
      </section>
    </div>
  );
};

export default ResourcesCenter;