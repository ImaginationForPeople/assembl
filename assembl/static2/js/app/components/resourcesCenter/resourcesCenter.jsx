// @flow
import React from 'react';
import { I18n } from 'react-redux-i18n';
import Header from '../common/header';
import ResourceBlock from './resourceBlock';

export type Media = {
  type: string,
  url: string
};

export type Resource = {
  title: string,
  text: string,
  image: Object,
  embedCode: string,
  doc: Object
};

type ResourcesCenterProps = {
  resources: Array<Resource>,
  headerBackgroundUrl: string
};

const ResourcesCenter = ({ resources, headerBackgroundUrl }: ResourcesCenterProps) => {
  const headerTitle = I18n.t('resourcesCenter.headerTitle');
  return (
    <div className="resources-center">
      <Header title={headerTitle} imgUrl={headerBackgroundUrl} />
      <section>
        {resources.map((resource, index) => {
          const { title, text, embedCode, image, doc } = resource;
          return (
            <ResourceBlock title={title} text={text} embedCode={embedCode} image={image} doc={doc} index={index} key={index} />
          );
        })}
      </section>
    </div>
  );
};

export default ResourcesCenter;