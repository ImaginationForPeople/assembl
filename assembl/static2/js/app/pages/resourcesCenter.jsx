import React from 'react';
import { I18n } from 'react-redux-i18n';
import { compose, graphql } from 'react-apollo';
import { connect } from 'react-redux';
import ResourcesQuery from '../graphql/ResourcesQuery.graphql';
import ResourcesCenterPageQuery from '../graphql/ResourcesCenterPage.graphql';
import withLoadingIndicator from '../components/common/withLoadingIndicator';
import ResourcesCenter from '../components/resourcesCenter';

const ResourcesCenterContainer = ({ data, resourcesCenterHeaderUrl, resourcesCenterTitle }) => {
  return <ResourcesCenter {...data} headerBackgroundUrl={resourcesCenterHeaderUrl} headerTitle={resourcesCenterTitle} />;
};

const mapStateToProps = (state) => {
  return {
    debate: state.debate,
    lang: state.i18n.locale
  };
};

export default compose(
  connect(mapStateToProps),
  graphql(ResourcesCenterPageQuery, {
    props: ({ data, ownProps }) => {
      const defaultHeaderImage = ownProps.debate.debateData.headerBackgroundUrl || '';
      if (data.loading) {
        return {
          loading: true
        };
      }
      if (data.error) {
        return {
          hasErrors: true
        };
      }

      const title = data.resourcesCenter.title;
      const headerImage = data.resourcesCenter.headerImage;
      return {
        resourcesCenterTitle: title || I18n.t('resourcesCenter.defaultHeaderTitle'),
        resourcesCenterHeaderUrl: headerImage ? headerImage.externalUrl : defaultHeaderImage
      };
    }
  }),
  graphql(ResourcesQuery),
  withLoadingIndicator()
)(ResourcesCenterContainer);