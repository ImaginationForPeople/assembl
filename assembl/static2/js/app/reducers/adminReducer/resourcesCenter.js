// @flow
import { combineReducers } from 'redux';
import type ReduxAction from 'redux';
import { fromJS, List, Map } from 'immutable';

import {
  type Action,
  CREATE_RESOURCE,
  DELETE_RESOURCE,
  UPDATE_RESOURCE_DOCUMENT,
  UPDATE_RESOURCE_EMBED_CODE,
  UPDATE_RESOURCE_IMAGE,
  UPDATE_RESOURCE_TEXT,
  UPDATE_RESOURCE_TITLE,
  UPDATE_RESOURCES,
  UPDATE_RC_PAGE_TITLE,
  UPDATE_RC_HEADER_IMAGE,
  UPDATE_RC_PAGE
} from '../../actions/actionTypes';
import { updateInLangstringEntries } from '../../utils/i18n';

const initialPage = Map({
  hasChanged: false,
  titleEntries: List(),
  headerImage: Map({
    externalUrl: '',
    mimeType: '',
    title: ''
  })
});
export const page = (state: Map = initialPage, action: ReduxAction<Action>) => {
  switch (action.type) {
  case UPDATE_RC_PAGE_TITLE:
    return state.update('titleEntries', updateInLangstringEntries(action.locale, fromJS(action.value))).set('hasChanged', true);
  case UPDATE_RC_HEADER_IMAGE:
    return state
      .setIn(['headerImage', 'externalUrl'], action.value)
      .setIn(['headerImage', 'mimeType'], action.value.type)
      .set('hasChanged', true);
  case UPDATE_RC_PAGE: {
    let newState = state;
    if (action.headerImage) {
      newState = newState
        .setIn(['headerImage', 'externalUrl'], action.headerImage.externalUrl)
        .setIn(['headerImage', 'mimeType'], action.headerImage.mimeType);
    }

    if (action.titleEntries) {
      newState = newState.set('titleEntries', fromJS(action.titleEntries));
    }

    return newState.set('hasChanged', false);
  }
  default:
    return state;
  }
};

export const resourcesHaveChanged = (state: boolean = false, action: ReduxAction<Action>) => {
  switch (action.type) {
  case CREATE_RESOURCE:
  case DELETE_RESOURCE:
  case UPDATE_RESOURCE_DOCUMENT:
  case UPDATE_RESOURCE_EMBED_CODE:
  case UPDATE_RESOURCE_IMAGE:
  case UPDATE_RESOURCE_TEXT:
  case UPDATE_RESOURCE_TITLE:
    return true;
  case UPDATE_RESOURCES:
    return false;
  default:
    return state;
  }
};

export const resourcesInOrder = (state: List<number> = List(), action: ReduxAction<Action>) => {
  switch (action.type) {
  case CREATE_RESOURCE:
    return state.push(action.id);
  case UPDATE_RESOURCES:
    return List(action.resources.map(r => r.id));
  default:
    return state;
  }
};

// TODO: declare Map shape for ResourceItem (it seems not possible for now:
// https://stackoverflow.com/questions/37033339/how-to-describe-immutable-js-map-shape-with-flow)
// maybe we should use Record instead of Map for this kind of stuff
//
// type LangString = {
//   localeCode: string,
//   value: string
// };
//
// type ResourceItem = {
//   toDelete: boolean,
//   isNew: boolean,
//   doc: { externalUrl: string },
//   id: string,
//   img: { externalUrl: string, mimeType: string },
//   titleEntries: List<LangString>,
//   textEntries: List<LangString>,
//   embedCode: string
// };
const defaultResourceDoc = Map({
  externalUrl: ''
});
const defaultResourceImage = Map({
  externalUrl: '',
  mimeType: ''
});
const defaultResource = Map({
  toDelete: false,
  isNew: true,
  doc: defaultResourceDoc,
  img: defaultResourceImage,
  titleEntries: List(),
  textEntries: List(),
  embedCode: ''
});
export const resourcesById = (state: Map<string, Map> = Map(), action: ReduxAction<Action>) => {
  switch (action.type) {
  case CREATE_RESOURCE:
    return state.set(action.id, defaultResource.set('id', action.id).set('order', action.order));
  case DELETE_RESOURCE:
    return state.setIn([action.id, 'toDelete'], true);
  case UPDATE_RESOURCE_DOCUMENT:
    return state.setIn([action.id, 'doc', 'externalUrl'], action.value);
  case UPDATE_RESOURCE_EMBED_CODE:
    return state.setIn([action.id, 'embedCode'], action.value);
  case UPDATE_RESOURCE_IMAGE:
    return state
      .setIn([action.id, 'img', 'externalUrl'], action.value)
      .setIn([action.id, 'img', 'mimeType'], action.value.type);
  case UPDATE_RESOURCE_TEXT:
    return state.updateIn([action.id, 'textEntries'], updateInLangstringEntries(action.locale, fromJS(action.value)));
  case UPDATE_RESOURCE_TITLE:
    return state.updateIn([action.id, 'titleEntries'], updateInLangstringEntries(action.locale, action.value));
  case UPDATE_RESOURCES: {
    let newState = Map();
    action.resources.forEach((resource, idx) => {
      const resourceInfo = Map({
        toDelete: false,
        isNew: false,
        order: idx + 1,
        doc: resource.doc ? fromJS(resource.doc) : defaultResourceDoc,
        id: resource.id,
        img: resource.image ? fromJS(resource.image) : defaultResourceImage,
        titleEntries: fromJS(resource.titleEntries),
        textEntries: fromJS(resource.textEntries),
        embedCode: resource.embedCode
      });

      newState = newState.set(resource.id, resourceInfo);
    });

    return newState;
  }
  default:
    return state;
  }
};

export default combineReducers({
  page: page,
  resourcesHaveChanged: resourcesHaveChanged,
  resourcesInOrder: resourcesInOrder,
  resourcesById: resourcesById
});