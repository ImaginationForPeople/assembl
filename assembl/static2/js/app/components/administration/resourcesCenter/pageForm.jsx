// @flow
import React from 'react';
import { connect } from 'react-redux';
import { I18n, Translate } from 'react-redux-i18n';
import { FormGroup } from 'react-bootstrap';

import FileUploader from '../../common/fileUploader';
import FormControlWithLabel from '../../common/formControlWithLabel';
import { updateResourcesCenterPageTitle, updateResourcesCenterHeaderImage } from '../../../actions/adminActions/resourcesCenter';
import { getEntryValueForLocale } from '../../../utils/i18n';

type PageFormProps = {
  handleHeaderImageChange: Function,
  handlePageTitleChange: Function,
  headerMimeType: string,
  headerUrl: File | string,
  selectedLocale: string,
  title: string
};

const DumbPageForm = ({
  handleHeaderImageChange,
  handlePageTitleChange,
  headerMimeType,
  headerUrl,
  selectedLocale,
  title
}: PageFormProps) => {
  const titleLabel = I18n.t('administration.resourcesCenter.pageTitleLabel');
  const headerImageFieldName = 'header-image';
  return (
    <div>
      <FormControlWithLabel key={selectedLocale} label={titleLabel} onChange={handlePageTitleChange} type="text" value={title} />
      <FormGroup>
        <label htmlFor={headerImageFieldName}>
          <Translate value="administration.resourcesCenter.headerImageLabel" />
        </label>
        <FileUploader
          mimeType={headerMimeType}
          name={headerImageFieldName}
          fileOrUrl={headerUrl}
          handleChange={handleHeaderImageChange}
        />
      </FormGroup>
      <div className="separator" />
    </div>
  );
};

const mapStateToProps = (state, { selectedLocale }) => {
  const page = state.admin.resourcesCenter.page;
  return {
    title: getEntryValueForLocale(page.get('titleEntries'), selectedLocale, ''),
    headerFilename: page.getIn(['headerImage', 'title']),
    headerMimeType: page.getIn(['headerImage', 'mimeType']),
    headerUrl: page.getIn(['headerImage', 'externalUrl'])
  };
};

const mapDispatchToProps = (dispatch, { selectedLocale }) => {
  return {
    handleHeaderImageChange: (value) => {
      return dispatch(updateResourcesCenterHeaderImage(value));
    },
    handlePageTitleChange: (e) => {
      return dispatch(updateResourcesCenterPageTitle(selectedLocale, e.target.value));
    }
  };
};

export { DumbPageForm };

export default connect(mapStateToProps, mapDispatchToProps)(DumbPageForm);