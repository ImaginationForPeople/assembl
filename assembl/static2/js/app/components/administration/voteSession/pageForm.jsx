// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';

import SectionTitle from '../sectionTitle';
import FormControlWithLabel from '../../common/formControlWithLabel';
import Helper from '../../common/helper';
import { getEntryValueForLocale } from '../../../utils/i18n';
import { updateVoteSessionPagePropositionsTitle } from '../../../actions/adminActions/voteSession';

type Props = {
  propositionSectionTitle: string,
  editLocale: string,
  handlePropositionSectionTitleChange: Function
};

const DumbPageForm = ({ propositionSectionTitle, editLocale, handlePropositionSectionTitleChange }: Props) => {
  const editLocaleInUppercase = editLocale.toUpperCase();
  const propositionSectionTitlePh = `${I18n.t('administration.ph.propositionSectionTitle')} ${editLocaleInUppercase}`;
  return (
    <div className="admin-box">
      <SectionTitle title={I18n.t('administration.voteSession.0')} annotation={I18n.t('administration.annotation')} />
      <div className="admin-content">
        <div className="form-container">
          <Helper
            label={I18n.t('administration.proposalSectionTitle')}
            helperUrl="/static2/img/helpers/helper3.png"
            helperText={I18n.t('administration.helpers.voteSessionProposalSection')}
            classname="title"
          />
          <FormControlWithLabel label={propositionSectionTitlePh} onChange={handlePropositionSectionTitleChange} required type="text" value={propositionSectionTitle} />
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state, { editLocale }) => {
  const voteSession = state.admin.voteSession.page;
  return {
    propositionSectionTitle: getEntryValueForLocale(voteSession.get('propositionsSectionTitleEntries'), editLocale)
  };
};

const mapDispatchToProps = (dispatch, { editLocale }) => ({
  handlePropositionSectionTitleChange: e => dispatch(updateVoteSessionPagePropositionsTitle(editLocale, e.target.value))
});

export { DumbPageForm };

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DumbPageForm);