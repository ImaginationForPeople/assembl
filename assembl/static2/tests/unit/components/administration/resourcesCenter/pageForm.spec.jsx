import React from 'react';
import renderer from 'react-test-renderer';

import { DumbPageForm } from '../../../../../js/app/components/administration/resourcesCenter/pageForm';

describe('DumbPageForm component', () => {
  it('should render a form to update the resources center page title and header', () => {
    const handleHeaderImageChangeSpy = jest.fn(() => {});
    const handlePageTitleChangeSpy = jest.fn(() => {});
    const props = {
      handleHeaderImageChange: handleHeaderImageChangeSpy,
      handlePageTitleChange: handlePageTitleChangeSpy,
      headerMimeType: 'image/jpeg',
      headerUrl: 'http://www.example.com/documents/myimage/data',
      selectedLocale: 'fr',
      title: 'Centre de ressources'
    };
    const component = renderer.create(<DumbPageForm {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});