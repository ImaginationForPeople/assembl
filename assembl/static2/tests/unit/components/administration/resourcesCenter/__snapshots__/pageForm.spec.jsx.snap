// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DumbPageForm component should render a form to update the resources center page title and header 1`] = `
<div>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor={undefined}
    >
      Missing translation: administration.resourcesCenter.pageTitleLabel
    </label>
    <input
      className="form-control"
      id={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Missing translation: administration.resourcesCenter.pageTitleLabel"
      type="text"
      value="Centre de ressources"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="header-image"
    >
      <span
        className={undefined}
        style={undefined}
      >
        Missing translation: administration.resourcesCenter.headerImageLabel
      </span>
    </label>
    <div>
      <button
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className={undefined}
          style={undefined}
        >
          Missing translation: common.uploadButton
        </span>
      </button>
      <div
        className="preview"
      >
        <img
          alt="preview"
          src="http://www.example.com/documents/myimage/data"
        />
      </div>
      <input
        className="hidden"
        name="header-image"
        onChange={[Function]}
        type="file"
      />
    </div>
  </div>
  <div
    className="separator"
  />
</div>
`;
